numba/tests/test_np_functions.py

```diff
@@ -4862,6 +4867,35 @@ def not_literal_axis(a, i, axis):
             gen(0)(arr2d, np.ones((2, 3), dtype=np.uint64))
         self.assertIn("dimensions don't match", str(raises.exception))
 
+    def test_nan_to_num(self):
+        values = [
+            np.nan,
+            1.0,
+            np.asarray([0.1, 1.0, 0.4]),
+            np.asarray([[0.1, 1.0, 0.4], [0.4, 1.2, 4.0]]),
+            np.asarray([0.1, np.nan, 0.4]),
+            np.asarray([[0.1, np.nan, 0.4], [np.nan, 1.2, 4.0]]),
+        ]
+        nans = [0.0, 4.2]
```

Comment:
Could you copy the tests used by NumPy? You can find them [here](https://github.com/numpy/numpy/blob/8ff45c5bb520db04af8720bf1d34a392a8d2561a/numpy/lib/tests/test_type_check.py#L350-L452). Not all tests will work with Numba. Copy the ones that will work with your changes.